cloud/WFLY-11824_wildfly_operator_kubernetes.adoc

```diff
@@ -0,0 +1,126 @@
+= WildFly Operator for Kubernetes/OpenShift
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Managing an application running on WildFly in a cloud environment can be a complex task. Things like ensuring containers have access to necessary resources, scaling up and scale down efficiently, integrating with outside services and handling rollout of new versions are all complex tasks that require operation knowledge. An https://coreos.com/blog/introducing-operators.html[Operator] is a specialized form of Kubernetes / OpenShift controller that is meant to encode that kind of operation knowledge in software, allowing users to more easily manage their cloud applications. The goal of this feature is to develop and Operator for managing WildFly-based applications.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-11824[WFLY-11824]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1149[EAP7-1149]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+ * TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
+[ ] QE
+
+=== Affected Projects or Components
+
+ * https://github.com/wildfly/wildfly[WildFly]
+ * https://github.com/openshift-s2i/s2i-wildfly[WildFly s2i]
+
+
+=== Other Interested Projects
+
+ * https://github.com/jboss-dockerfiles/wildfly[WildFly Docker image]
+ * Keycloak / Red Hat SSO
+ * RHPAM
+
+== Requirements
+
+
+=== Hard Requirements
+
+==== Create a Custom Resource Definition (CRD) for WildFly.
+
+Provide a Custom Resource Definition (CRD) for an application based on a WildFly application server.
+
+The CRD will allow the user to:
+
+* Specify the name an application image that incorporates a WildFly server and an deployment that will be deployed by that server.
+** The name can include either a fixed or floating tag.
+* Specify an optional ConfigMap where the WildFly standalone configuration can be read from.
+* Specify the desired number of replicas of that image
+* Specify whether the replicas are intended to form a High Availability cluster (default is true).
+* Specify whether the containers are intended to be 'stateful'
+** default is to be determined (might be inferred from the server configuration)
+* Specify information for interfaces for which a Service and Route should be established
+** perhaps expose 8080 by default with an option to disable that
```

Comment:
In this case, if only allowing secure access, certificate management need to be considered as part of the operator.